woocommerce/woocommerce
# [Settings] Register and render internal slotFill

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Register and render slotFills from our own codebase. This task is normally [handled in wc-admin](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/client/embedded-body-layout/render-embedded-layout.tsx#L112) but we avoid loading the wc-admin app.

To use `useActiveRoute` it must be called from a component within `RouterProvider`, so that needed to be exported from the package and used at the root level. This allows re-rendering of slotFills when the tab or section change.

SlotFills generated by extensions won't be rendered until https://github.com/woocommerce/woocommerce/issues/53860 is handled.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add/Enable Gutenberg version `19.8.0` or earlier.
2. Turn on the `settings` feature flag
3. WooCommerce > Settings > Site visibility. The launch your store site visibility settings should appear but they will be non functional.
4. WooCommerce > Settings > Tax. See the error warning if applicable.
<img width="647" alt="image" src="https://github.com/user-attachments/assets/6993f6ad-f26f-4375-b98d-18e2c50e213d" />

5. WooCommerce > Settings > Payments. On a fresh site, this slotFill won't render anything but confirm no error messages appear.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Register and render internal slotFills in the new Settings.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
